src/spikeinterface/sortingcomponents/matching/wobble.py

```diff
@@ -0,0 +1,929 @@
+        """
+        visible_channels = spatially_mask_templates(templates, params.visibility_threshold)
+        unit_overlap = template_overlaps(visible_channels, params.jitter_factor)
+        sparsity = cls(visible_channels=visible_channels, unit_overlap=unit_overlap)
+        return sparsity
+
+
+def spatially_mask_templates(templates, visibility_threshold):
+    """Determine which channels are 'visible' for each template, based on peak-to-peak amplitude, and
+    set the template to 0 for 'invisible' channels.
+
+    Parameters
+    ----------
+    templates : ndarray (num_templates, num_samples, num_channels)
+        Spike template waveforms.
+    visibility_thresold : float
+        Minimum peak amplitude to determine channel sparsity for a given unit. Units depend on the underlying units
+        of voltage trace and templates.
+
+    Returns
+    -------
+    visible_channels : ndarray (num_units, num_channels)
+        visible_channels[unit, channel] is True if the unit's template has sufficient amplitude on that channel.
+    """
+    visible_channels = np.ptp(templates, axis=1) > visibility_threshold
+    invisible_channels = np.logical_not(visible_channels)
+    for i in range(templates.shape[0]):
+        templates[i, :, invisible_channels[i, :]] = 0.0
+    return visible_channels
+
+
+def template_overlaps(visible_channels, jitter_factor):
+    """Finds overlapping templates based on peak-to-peak amplitude at different channels.
+
+    Parameters
+    ----------
+    visible_channels : ndarray (num_units, num_channels)
+        visible_channels[unit, channel] is True if the unit's template has sufficient amplitude on that channel.
+    jitter_factor : int
+        Number of upsampled jittered templates for each distinct provided template.
+
+    Returns
+    -------
+    unit_overlap : ndarray (num_jittered, num_jittered)
+        unit_overlap[i, j] is True if there exists at least one channel on which both template i and template j are
+        visible.
+    """
+    unit_overlap = np.sum(np.logical_and(visible_channels[:, np.newaxis, :], visible_channels[np.newaxis, :, :]), axis=2)
+    unit_overlap = unit_overlap > 0
+    unit_overlap = np.repeat(unit_overlap, jitter_factor, axis=0)
+    return unit_overlap
+
+
+@dataclass
+class TemplateData:
+    """Data derived from the set of templates.
+
+    Parameters
+    ----------
+    compressed_templates : (ndarray, ndarray, ndarray, ndarray)
+        Templates compressed by singular value decomposition into temporal, singular, spatial, and upsampled_temporal
+        components.
+    pairwise_convolution : list[ndarray]
+        For each jittered template, pairwise_convolution of that template with each other overlapping template.
+    norm : ndarray (num_templates,)
+        Magnitude of each template for normalization.
+    """
+    compressed_templates : Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]
```

Comment:
Question: is it necessary to keep this as a tuple?
Can you keep each of the attributes exposed as attributes (spatial, temporal, etc).

Reply:
all of the components of the compressed templates almost always are needed together, so it has been easier to just pass them around together in a tuple.  But, just now, I also exposed them as attributes directly in case one might need to access only one at a time for some reason (it also allows more detailed documentation).